src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -389,10 +384,10 @@ export class NotebookService extends Disposable implements INotebookService {
 	private async doWithProvider<T>(providerId: string, op: (provider: INotebookProvider) => Thenable<T>): Promise<T> {
 		// Make sure the provider exists before attempting to retrieve accounts
 		let provider: INotebookProvider = await this.getProviderInstance(providerId);
-		return op(provider);
+		return await op(provider);
```

Comment:
since 'op' is thenable it should be await to avoid any weird timing issue. So this is a bug fix that may not have surfaced into an issue.

Reply:
This is doing nothing except adding extra overhead. If a user wasn't .then/await'ing the result then this await is just causing it to need to re-wrap the result in a Promise.

Now if callers weren't properly awaiting the call to doWithProvider then that's a separate issue, but this change should be reverted.

Reply:
The only time you'd want to do this is if you want to handle potential exceptions here instead of at the caller.

Reply:
reverted